{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3b233961",
   "metadata": {},
   "outputs": [],
   "source": [
    "Q1. Describe three applications for exception processing.\n",
    "Q2. What happens if you don't do something extra to treat an exception?\n",
    "Q3. What are your options for recovering from an exception in your script?\n",
    "Q4. Describe two methods for triggering exceptions in your script.\n",
    "Q5. Identify two methods for specifying actions to be executed at termination time, regardless of whether or not an exception exists."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2c5a6f49",
   "metadata": {},
   "outputs": [],
   "source": [
    "Q1. Describe three applications for exception processing.\n",
    "\n",
    "ans:\n",
    "    Exception handling is done for runtime error not for compile time error.\n",
    "    \n",
    "    Three application for exception processing are:\n",
    "        1.arithmetic execption \n",
    "        2.file not found exception\n",
    "        3.array out of range exception"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e75c3fca",
   "metadata": {},
   "outputs": [],
   "source": [
    "Q2. What happens if you don't do something extra to treat an exception?\n",
    "\n",
    "ans:\n",
    "    If Exceptions are not handled flow of program will be broken during the run time which might lead to a abnormal termination of the program.\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4a9d8a22",
   "metadata": {},
   "outputs": [],
   "source": [
    "Q3. What are your options for recovering from an exception in your script?\n",
    "\n",
    "ans:\n",
    "    The options for recovering from an exception in our script is by writing a program using exception handling statemnets i.e., try and except\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "89b19a61",
   "metadata": {},
   "outputs": [],
   "source": [
    "Q4. Describe two methods for triggering exceptions in your script.\n",
    "\n",
    "ans:\n",
    "raise and assert are two methods that can be used to trigger manual exceptions in your script.\n",
    "\n",
    "raise method triggers an exception if condition provided to it turns out to be True.\n",
    "assert will let the program to continue execution if condition provided to it turns out to be True else exception will be raised\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8352cc25",
   "metadata": {},
   "outputs": [],
   "source": [
    "Q5. Identify two methods for specifying actions to be executed at termination time, regardless of whether or not an exception exists.\n",
    "\n",
    "ans:\n",
    "    else and finally are the two methods for specifying actions to be executed at termination time, regardless of whether or not an exception exists.\n",
    "    "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
